### ISSUE TITLE
Test Failure: expected IOException is wrapped in Akka Stream exception

### ISSUE THREAD

**Issue by ignasi35** [playframework/playframework#9491]
Seen in travis:

```
[info] MultipartBodyParserSpec
[info] Multipart body parser should
[error]   x return an error if temporary file creation fails (130 ms)
[error]    Expected: java.io.IOException. Got: akka.stream.IOOperationIncompleteException: java.io.IOException: out of disk space instead 
[error]    
[error]     The  IOOperationIncompleteException stacktrace is
[error]    
[error]    akka.stream.impl.io.FileOutputStage$$anon$1.onPush(FileOutputStage.scala:67)
[error]    akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:526)
[error]    akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:482)
[error]    akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:377)
[error]    akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:610)
[error]    akka.stream.impl.fusing.ActorGraphInterpreter$SimpleBoundaryEvent.execute(ActorGraphInterpreter.scala:47)
[error]    akka.stream.impl.fusing.ActorGraphInterpreter$SimpleBoundaryEvent.execute$(ActorGraphInterpreter.scala:43)
[error]    akka.stream.impl.fusing.ActorGraphInterpreter$BatchingActorInputBoundary$OnNext.execute(ActorGraphInterpreter.scala:86)
[error]    akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:585)
[error]    akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:753)
[error]    akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:768)
[error]    akka.actor.Actor.aroundReceive(Actor.scala:532)
[error]    akka.actor.Actor.aroundReceive$(Actor.scala:530)
[error]    akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:675)
[error]    akka.actor.ActorCell.receiveMessage(ActorCell.scala:571)
[error]    akka.actor.ActorCell.invoke(ActorCell.scala:541)
[error]    akka.dispatch.Mailbox.processMailbox(Mailbox.scala:269)
[error]    akka.dispatch.Mailbox.run(Mailbox.scala:230)
[error]    java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[error]    java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[error]    java.lang.Thread.run(Thread.java:748) (MultipartBodyParserSpec.scala:63)
[info] Total for specification MultipartBodyParserSpec
[info] Finished in 178 ms
[info] 1 example, 1 failure, 0 error
[
```

This is probably a change in upstream Akka and this exception only happens if/When the [CRON job](https://travis-ci.com/playframework/playframework/jobs/218060206) uses an Akka snapshot.

**Comment by ignasi35** [comment]
Note this is a consistent failure (not a flaky test).

**Comment by ignasi35** [comment]
related to https://github.com/playframework/playframework/pull/8917#issuecomment-452032487

related to https://github.com/akka/akka/issues/26187 and https://github.com/akka/akka/pull/26811 (which deprecates `IOResult`).

**Comment by dwijnand** [comment]
Fixed in #9506
